mqtt_client.c

```diff
@@ -990,6 +992,30 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t deliver_suback(esp_mqtt_client_handle_t client)
+{
+    uint8_t *msg_buf = client->mqtt_state.in_buffer;
+    size_t msg_total_len = client->mqtt_state.message_length;
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
+    int total_data_len = msg_total_len - 4;
```

Comment:
Suggest setting zero length if `msg_total_len - 4` (understand that this should never happen and you're setting the data ptr to NULL if it does, but still like to avoid negative lengths in user events)

Reply:
willdo, in mqtt_msg.c length will be checked and set, returned pointer nulled if zero.